repository_service_tuf/cli/artifact/download.py

```diff
@@ -0,0 +1,263 @@
+
+
+def decode_trusted_root(root) -> str:
+    root_encoded = bytes(root, "utf-8")
+    return base64.b64decode(root_encoded).decode("utf-8")
+
+
+def build_metadata_dir(metadata_url: str) -> str:
+    """build a unique and reproducible directory name for the repository url"""
+    name = sha256(metadata_url.encode()).hexdigest()[:8]
+    return f"{Path.home()}/.local/share/rstuf/{name}"
+
+
+def init_tofu(metadata_url: str, root_url: Optional[str]) -> bool:
+    """Initialize local trusted metadata (Trust-On-First-Use) and create a
+    directory for downloads"""
+    metadata_dir = build_metadata_dir(metadata_url)
+
+    os.makedirs(metadata_dir, exist_ok=True)
+
+    if not root_url:
+        root_url = f"{metadata_url}/1.root.json"
+    try:
+        parsed_root_url = urlparse(root_url)
+        if parsed_root_url.scheme and parsed_root_url.netloc:
+            request.urlretrieve(root_url, f"{metadata_dir}/root.json")  # nosec
+        else:
+            console.print(
+                f"Failed to parse {root_url}: ",
+                f"{parsed_root_url.scheme}, {parsed_root_url.netloc}",
+            )
+    except OSError:  # pragma: no cover
+        console.print(f"Failed to download initial root from {root_url}")
+        return False  # pragma: no cover
+
+    console.print(
+        f"Trust-on-First-Use: Initialized new root in {metadata_dir}"
+    )
+    return True
+
+
+def download_artifact(
+    metadata_url: Optional[str],
+    artifacts_url: Optional[str],
+    artifact_name: str,
+    root: Optional[str],
+    directory_prefix: Optional[str],
+) -> bool:
+    if metadata_url is None:
+        console.print("Please specify metadata url")
+        return False
+    metadata_dir = build_metadata_dir(metadata_url)
+
+    if artifacts_url is None:
+        console.print("Please specify artifacts url")
+        return False
+
+    if not os.path.isfile(f"{metadata_dir}/root.json"):
+        console.print(
+            "Trusted local root not found. Using 'tofu' to "
+            "Trust-On-First-Use or copy trusted root metadata to "
+            f"{metadata_dir}/root.json"
+        )
+        ok = init_tofu(metadata_url, root)
+        if not ok:
+            console.print(
+                f"Trusted local root not found in {metadata_url} - "
+                "`tofu` was not successful"
+            )
+            return False
+
+    print(f"Using trusted root in {metadata_dir}")
+
+    if directory_prefix:
+        # Annotating as it's covered, but codecov doesn't
+        # recognize that properly
+        download_dir = directory_prefix  # pragma: no cover
+    else:
+        download_dir = os.getcwd() + "/downloads"
+
+    if not os.path.isdir(download_dir):
+        os.mkdir(download_dir)
+
+    try:
+        updater = Updater(
+            metadata_dir=metadata_dir,
+            metadata_base_url=metadata_url,
+            target_base_url=artifacts_url,
+            target_dir=download_dir,
+        )
```

Comment:
Currently, python-tuf adds sha256 string as a prefix to the artifact name.
There is a configuration object for that based on `UpdaterConfig` object from python-tuf 
that we can use `prefix_targets_with_hash`: https://github.com/theupdateframework/python-tuf/blob/bbe2ca84a9ccd1397ab3229ada2ac2154d60ddad/tuf/ngclient/config.py#L35.
```suggestion
        updater = Updater(
            metadata_dir=metadata_dir,
            metadata_base_url=metadata_url,
            target_base_url=artifacts_url,
            target_dir=download_dir,
            config=UpdaterConfig(prefix_targets_with_hash=False)
        )
```